src/plugins/onClick.js

```diff
@@ -0,0 +1,52 @@
+let handleFun = null;
+let displayEl = null;
+
+function onClick(callback) {
+  displayEl = this.displayEl;
+  handleFun = handleEvent.bind(this, callback);
+  displayEl.addEventListener('click', handleFun);
+}
+
+function handleEvent(callback, event) {
+  const coords = findCoodrs(displayEl, event.target);
+  if (!coords) {
+    return;
+  }
+
+  const result = {
+    x: coords.x,
+    y: coords.y,
+    checkbox: event.target,
+  }
+  
+  if (typeof callback == 'function') {
+    callback(result);
+  } else if ('handleEvent' in callback && typeof callback.handleEvent == 'function') {
+    callback.handleEvent(result);
+  } else {
+    throw new TypeError('Callback should be a function or an EventListener object');
+  }
+}
+
+function findCoodrs(root, target) {
+  for (let y = 0; y < root.children.length; y += 1) {
+    const div = root.children[y];
+    for (let x = 0; x < div.children.length; x += 1) {
+      const checkbox = div.children[x];
+      if (checkbox === target) {
+        return { x, y };
+      }
+    }
+  }
+  return null;
+}
+
+function cleanUp() {
+  displayEl.removeEventListener('click', handleFun);
```

Comment:
This works great, and is exactly what `cleanUp()` was designed for. 👍

Reply:
Two more things I wanted to mention:
- Kudos for using event delegation, to make sure that we only needed one event for the whole grid. 🏆 
- I didn't event know about the [`EventListener` object interface](https://developer.mozilla.org/en-US/docs/Web/API/EventListener), but you added support for it and a nice error message. I learned something new today. 😄

Reply:
I didn't know about thet `EventListener` object either for a long time. I learned about it few months ago when I was creating an `addEventListener` implementation in TypeScript.
I'm glad that you've learned something from my code :)